### PR TITLE
External Linking facilities

### PR DESCRIPTION
**Pull request by gnunicorn**
Uses the new [link share generator service](https://github.com/acterglobal/synapse-super-invites/pull/8) (deployed on m-1 already) to generate external links. Also provides the facilities to generate the internal links.

Implemented for:
 - Pin
 - TaskList
 - CalendarEvent

Also exposes a new `ref_details()`-generator feature on each of them (which is used internally), which can be used as a system-wide reference object to attach to other objects (e.g. boosts) directly. Which can also be used as a new attachment type on all attachment-able objects (see generated api). 

Brings end-2-end-tests and includes the super invites branch in the tests already

### CODE REVIEW

File: native/test/src/tests/news.rs [hunk 1]
```diff
@@ -862,3 +862,66 @@ async fn multi_news_read_receipt_test() -> Result<()> {
 
     Ok(())
 }
+
+// #[tokio::test]
```
Comment (gtalha07) [inline native/test/src/tests/news.rs:866]: This doesn't work yet ? Then we might have it removed ?

File: native/test/src/tests/thumbnail.rs [hunk 2]
```diff
@@ -518,7 +518,7 @@ async fn image_attachment_can_support_thumbnail() -> Result<()> {
     assert_eq!(attachment.event_id(), attachment_id);
     assert_eq!(attachment.type_str(), "image");
 
-    let msg_content = attachment.msg_content();
+    let msg_content = attachment.msg_content().unwrap();
```
Comment (gtalha07) [inline native/test/src/tests/thumbnail.rs:521]: is this safe? Better give ```.expect()``` here for tests easier debugging?

**Review by gtalha07 — APPROVED** [review]
Looks good to me. Just some minor queries at your disposal :)

File: native/test/src/tests/thumbnail.rs (hunk 2 above)
Comment (gnunicorn) [reply native/test/src/tests/thumbnail.rs:521]: in tests we get full tracebacks including line-numbers, `unwrap` is fine here.

**Pull request merged**